os/arch/arm/src/armv8-m/up_exception.S

```diff
@@ -202,14 +202,14 @@ exception_common:
 
 	stmdb	sp!, {r2-r11,r14}		/* Save the remaining registers plus the SP/PRIMASK values */
 
-#ifndef CONFIG_ARCH_HIPRI_INTERRUPT
+#if !defined(CONFIG_ARCH_HIPRI_INTERRUPT) && !defined(CONFIG_ARCH_NESTED_INTERRUPT)
 	/* Disable interrupts, select the stack to use for interrupt handling
 	 * and call up_doirq to handle the interrupt
 	 */
 
 	cpsid	i						/* Disable further interrupts */
 
-#else
+#elif !defined(CONFIG_ARCH_NESTED_INTERRUPT)
```

Comment:
why can't we put CONFIG_ARCH_NESTED_INTERRUPT in #else?

Reply:
No.. First case if when nested and basepri are both undefined. Second case is when basepri is defined but nested is undefined. Third case below is for only nested being defined.

Reply:
@sangam-swami 
I modified according to your comments.